internal/common/tree.go

```diff
@@ -42,15 +43,53 @@ type DNSTreeNodeData struct {
 //   - Any node with the stopLabel and the propLabel:
 //   - Has the propLabel removed from itself and all it's children
 //   - Has the propLabel removed from any parent (or parent's parent) that has the label
-
 func PropagateStoppableLabel(node *DNSTreeNode, propLabel, value, stopLabel string) {
 	//propagate labels regardless of stop labels
-	propagateLabel(node, propLabel, value)
+	PropagateLabel(node, propLabel, value)
 
 	//propagate stop labels
 	resolveStops(node, propLabel, value, stopLabel)
 }
 
+func CopyLabel(label string, from, to *DNSTreeNode) {
+	toNode := FindNode(to, from.Name)
+	if toNode != nil {
+		// copy label values to any matching datasets
+		for _, fd := range from.Endpoints {
+			for _, td := range toNode.Endpoints {
+				if fd.Key() == td.Key() {
+					if fd.Labels == nil || fd.Labels[label] == "" {
+						if td.Labels != nil {
+							delete(td.Labels, label)
+						}
+					} else if fd.Labels[label] != "" {
```

Comment:
The "it could be easier" bit (ignorable):
Shouldn't it just be `else`? :thinking:

Reply:
:+1: